### PR TITLE
Release: 2.3.0 

### PR DESCRIPTION
**Pull request by jfy133**
### `Added`

- [#350](https://github.com/nf-core/mag/pull/350) - Adds support for CheckM as alternative bin completeness and QC tool (added by @jfy133 and @skrakau)
- [#353](https://github.com/nf-core/mag/pull/353) - Added the busco_clean parameter to optionally clean each BUSCO directory after a successful (by @prototaxites)
- [#361](https://github.com/nf-core/mag/pull/361) - Added the skip_clipping parameter to skip read preprocessing with fastp or adapterremoval. Running the pipeline with skip_clipping, keep_phix and without specifying a host genome or fasta file skips the FASTQC_TRIMMED process (by @prototaxites)
- [#365](https://github.com/nf-core/mag/pull/365) - Added CONCOCT as an additional (optional) binning tool (by @jfy133)
- [#366](https://github.com/nf-core/mag/pull/366) - Added CAT_SUMMARISE process and cat_official_taxonomy parameter (by @prototaxites)
- [#372](https://github.com/nf-core/mag/pull/372) - Allow CAT_DB to take an extracted database as well as a tar.gz file (by @prototaxites).
- [#380](https://github.com/nf-core/mag/pull/380) - Added support for saving processed reads (clipped, host removed etc.) to results directory (by @jfy133)
- [#394](https://github.com/nf-core/mag/pull/394) - Added GUNC for additional chimeric bin/contamination QC (added by @jfy13)

### `Changed`

- [#340](https://github.com/nf-core/mag/pull/340),[#368](https://github.com/nf-core/mag/pull/368),[#373](https://github.com/nf-core/mag/pull/373) - Update to nf-core 2.7.2 `TEMPLATE` (by @jfy133, @d4straub, @skrakau)
- [#373](https://github.com/nf-core/mag/pull/373) - Removed parameter `--enable_conda`. Updated local modules to new conda syntax and updated nf-core modules (by @skrakau)
- [#385](https://github.com/nf-core/mag/pull/385) - CAT also now runs on unbinned contigs as well as binned contigs (added by @jfy133)

### `Fixed`

- [#345](https://github.com/nf-core/mag/pull/345) - Bowtie2 mode changed to global alignment for ancient DNA mode (`--very-sensitive` mode) to prevent soft clipping at the end of reads when running in local mode. (by @maxibor)
- [#349](https://github.com/nf-core/mag/pull/349) - Add a warning that pipeline will reset minimum contig size to 1500 specifically MetaBAT2 process, if a user supplies below this threshold. (by @jfy133)
- [#352](https://github.com/nf-core/mag/pull/352) - Escape the case in the BUSCO module that BUSCO can just detect a root lineage but is not able to find any marker genes (by @alexhbnr)
- [#355](https://github.com/nf-core/mag/pull/355) - Include error code 21 for retrying with higher memory for SPAdes and hybridSPAdes (by @mglubber)

### `Dependencies`

| Tool      | Previous version | New version |
| --------- | ---------------- | ----------- |
| BUSCO     | 5.1.0            | 5.4.3       |
| BCFtools  | 1.14             | 1.16        |
| Freebayes | 1.3.5            | 1.3.6       |
| SAMtools  | 1.15             | 1.16.1      |

## PR checklist

- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
  - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/mag/tree/master/.github/CONTRIBUTING.md)
  - [ ] If necessary, also make a PR on the nf-core/mag _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [X] Usage Documentation in `docs/usage.md` is updated.
- [X] Output Documentation in `docs/output.md` is updated.
- [X] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: modules/nf-core/multiqc/main.nf
Comment (d4straub) [inline modules/nf-core/multiqc/main.nf:7]: modules/local/multiqc.nf uses 1.12, is is required to have 2 different versions?

**Review by d4straub — APPROVED** [review]
That are a lot of template changes... maybe we should indeed more often do minor releases right after template updates so that the release PRs are more functional changes.

LGTM.

File: modules/local/busco.nf
Comment (ggabernet) [inline modules/local/busco.nf:37]:
```suggestion
    def busco_clean = params.busco_clean ? "Y" : "N"
```

Comment (ggabernet) [reply modules/local/busco.nf:37]: could also be simplified for the above statement :)

File: modules/local/busco.nf
Comment (ggabernet) [inline modules/local/busco.nf:41]:
```suggestion
    def p = params.busco_reference ? "--lineage_dataset dataset/${db}": "--auto-lineage"
```

**Review by ggabernet — APPROVED** [review]
Looks good to me, just had a couple of commentss

File: modules/local/bin_summary.nf
Comment (jfy133) [reply modules/local/bin_summary.nf:3]: Don't know - I leave this to @skrakau as I'm not familiar with this module

File: modules/local/busco_summary.nf
Comment (jfy133) [reply modules/local/busco_summary.nf:6]: I would agree - again leaving to @skrakau to decide

File: modules/local/gtdbtk_summary.nf
Comment (jfy133) [reply modules/local/gtdbtk_summary.nf:7]: @skrakau (sorry for the spam, just making sure we don't forget anything)

File: modules/local/mag_depths_summary.nf
Comment (jfy133) [reply modules/local/mag_depths_summary.nf:6]: @skrakau

File: modules/nf-core/multiqc/main.nf
Comment (jfy133) [reply modules/nf-core/multiqc/main.nf:7]: no, you guys still have a 'custom' MultiQC module - this one isn't used, but I've left it to make template merges easier :)

File: modules/local/busco_plot.nf
Comment (jfy133) [reply modules/local/busco_plot.nf:7]: Good question, will review to check all are in sync with the latest version :+1:

**Comment by jfy133** [comment]
Waiting for a third review from @skrakau then we can go :)

File: modules/local/busco_plot.nf
Comment (skrakau) [reply modules/local/busco_plot.nf]: there was a [change ](https://gitlab.com/ezlab/busco/-/commit/f4744bf0479c562a8a9a6630f31584a286f75b39)in the used `generate_plot.py` script from BUSCO with v5.3.0, which would require some adjustments of the `short_summary` file names in our pipeline if I see it correctly. Currently they are called `"short_summary.domain.*"` or `"short_summary.specific_lineage.*"` (see [here](https://github.com/nf-core/mag/blob/5329440f663c54ce28c2362a8ba167ccb9b60e14/modules/local/busco.nf#L15)). 

We could rename the "mode" in the [`busco_plot.nf` script](https://github.com/nf-core/mag/blob/5329440f663c54ce28c2362a8ba167ccb9b60e14/modules/local/busco_plot.nf#L25) accordingly and rename it back afterwards.

Or we could just discard this process :D

Comment (jfy133) [reply modules/local/busco_plot.nf]: Discard is an option because it's not useful?

File: modules/local/bin_summary.nf
Comment (skrakau) [reply modules/local/bin_summary.nf:3]: Yes, that should be compatible

File: docs/output.md
Comment (skrakau) [inline docs/output.md:350]:
```suggestion
All the files and contigs in these folders will be assessed by QUAST and BUSCO, if the parameter `--postbinning_input` is not set to `refined_bins_only`.
```

?

File: docs/output.md
Comment (skrakau) [inline docs/output.md:506]:
```suggestion
GUNC will be run if specified with `--run_gunc` as a standalone, unless CheckM is also activated via `--qc_tool 'checkm'`, in which case GUNC output will be merged with the CheckM output using `gunc merge_checkm`.
```

File: docs/usage.md
Comment (skrakau) [inline docs/usage.md:74]:
```suggestion
nextflow run nf-core/mag --input samplesheet.csv --outdir <OUTDIR> -profile docker
```
The `--host_genome` parameter is not required for this pipeline and thus I would omit it here and keep a minimal example to run the pipeline with default parameters.

File: lib/WorkflowMag.groovy
Comment (skrakau) [inline lib/WorkflowMag.groovy:131]:
```suggestion
            log.warn '--skip_binqc and --gtdb are specified! GTDB-tk will be omitted because GTDB-tk bin classification requires bin filtering based on BUSCO or CheckM QC results to avoid GTDB-tk errors.'
```

File: lib/WorkflowMag.groovy
Comment (skrakau) [inline lib/WorkflowMag.groovy:107]:
```suggestion
        if (params.skip_binqc && params.binqc_tool == 'checkm') {
            log.error 'Both --skip_binqc and --binqc_tool \'checkm\' are specified! Invalid combination, please specify either --skip_binqc or --binqc_tool.'
            System.exit(1)
        }
        if (params.skip_binqc) {
```

If someone specifies CheckM and --skip_binqc, this should maybe better cause an error.

File: modules/local/bin_summary.nf
Comment (skrakau) [reply modules/local/bin_summary.nf:3]: not sure, from @d4straub it sounded in the past as if these plots are not particularly useful, not sure if others look into them. We can of course just adjust the script to keep it like it is, it's not super beautiful, but should work relatively easily.

Comment (d4straub) [reply modules/local/bin_summary.nf:3]: Those plots are typically too small (i.e. too many bins) to be informative, in my experience. I rather look at the final table, thats the most important output (besides sequences) for me.

File: modules/local/busco_plot.nf
Comment (d4straub) [reply modules/local/busco_plot.nf]: Those plots are typically too small (i.e. too many bins) to be informative, in my experience. I rather look at the final table, thats the most important output (besides sequences) for me.

File: modules/local/gtdbtk_summary.nf
Comment (jfy133) [reply modules/local/gtdbtk_summary.nf:7]: Addressed in : https://github.com/nf-core/mag/pull/399

File: modules/local/busco_plot.nf
Comment (jfy133) [reply modules/local/busco_plot.nf]: Addressed in : https://github.com/nf-core/mag/pull/399

**Review by skrakau — APPROVED** [review]
LGTM! Thanks @jfy133 :)

**Pull request merged**
